model/src/main/java/org/projectnessie/model/Reference.java

```diff
@@ -80,4 +81,19 @@ default void checkHash() {
   @JsonInclude(Include.NON_NULL)
   @Nullable
   ReferenceMetadata getMetadata();
+
+  @JsonIgnore
+  @Value.Redacted
+  ReferenceType getType();
+
+  /**
+   * The reference type, used inside REST API paths.
+   *
+   * <p>Enum values are intentionally lower-case, so this enum can be directly used as a REST API
+   * path parameter.
+   */
+  enum ReferenceType {
+    branch,
```

Comment:
Upper case enum name?

Reply:
See comment - intentionally lower case.

Reply:
Sorry missed that :facepalm:

Reply:
On the other hand, could we manually convert the path parameter (`String`) to the enum and uphold  the upper case naming convention? Would it be "cleaner" code-wise?

I'm a bit hesitant because the `model` module is not really specific to HTTP, but in this case we introduce a hidden dependency on something that is specific to HTTP. WDYT?

Reply:
Yup - updated that